### PR TITLE
Run repoclosure per package

### PR DESCRIPTION
**Pull request by ehelms**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.18
* [ ] 1.17
* [ ] 1.16

See Foreman's [plugin maintainer documentation](https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

**Comment by ehelms** [comment]
Will require https://github.com/theforeman/obal/pull/103

This will allow, for example:

```
obal repoclosure katello-host-tools
```

**Comment by ehelms** [comment]
Related CI updated -- https://github.com/theforeman/foreman-infra/pull/672

**Comment by ehelms** [comment]
The obal PR has been merged, this can be tested and reviewed

**Comment by komidore64** [comment]
this feature doesn't seem to give a nice message when one doesn't have the `repoclosure` command installed in their machine.

**Comment by komidore64** [comment]
i loaded up the obal changes from theforeman/obal#103, checked out this PR and tried to do a repoclosure check on katello-repoclosure-el7`. [no success so far](https://gist.github.com/komidore64/95a35a4c1951e4bd498e6dca5997f566). what did i mess up?

**Comment by komidore64** [comment]
i needed `yum-utils` instead of `dnf-utils`.

File: package_manifest.yaml [hunk 1]
```diff
@@ -11,6 +11,21 @@ foreman_packages:
...
+    repoclosure_lookaside_repos:
+      - el7-base
+      - el7-updates
+      - el7-epel
+      - el7-extras
+      - el7-scl
+      - el7-scl-sclo
+      - el7-scl-ruby
+      - el7-scl-v8
+      - el7-puppet-4
```
Comment (ekohl) [inline package_manifest.yaml]: I'd prefer puppet-5

File: package_manifest.yaml [hunk 2]
```diff
@@ -11,6 +11,21 @@ foreman_packages:
...
+      - el7-subscription-manager
```
Comment (ekohl) [inline package_manifest.yaml]: Should not be needed

File: package_manifest.yaml [hunk 3]
```diff
@@ -11,6 +11,21 @@ foreman_packages:
...
+      - el7-qpid
```
Comment (ekohl) [inline package_manifest.yaml]: Should not be needed

File: package_manifest.yaml [hunk 4]
```diff
@@ -11,6 +11,21 @@ foreman_packages:
...
     - koji-foreman
+    repoclosure_config: repoclosure/yum_el7.conf
+    repoclosure_lookaside_repos:
+      - el7-base
+      - el7-updates
+      - el7-epel
+      - el7-extras
+      - el7-scl
+      - el7-scl-sclo
+      - el7-scl-ruby
```
Comment (ekohl) [inline package_manifest.yaml]: Don't think this is needed anymore

File: package_manifest.yaml [hunk 5]
```diff
@@ -11,6 +11,21 @@ foreman_packages:
...
+      - el7-scl-v8
```
Comment (ekohl) [inline package_manifest.yaml]: Don't think this is needed anymore

File: package_manifest.yaml [hunk 6]
```diff
@@ -11,6 +11,21 @@ foreman_packages:
...
+      - el7-puppet-4
+      - el7-subscription-manager
+      - el7-qpid
+      - el7-tfm-ror51
```
Comment (ekohl) [inline package_manifest.yaml]: Haven't we renamed this to `el7-foreman-rails-nightly`?

File: package_manifest.yaml [hunk 7]
```diff
@@ -285,11 +300,31 @@ foreman_packages:
...
     - koji-katello
+    repoclosure_config: repoclosure/yum_el7.conf
+    repoclosure_lookaside_repos:
+      - el7-base
+      - el7-updates
+      - el7-epel
+      - el7-extras
+      - el7-scl
+      - el7-scl-sclo
+      - el7-scl-ruby
```
Comment (ekohl) [inline package_manifest.yaml]: See comments on `foreman_packages`

File: package_manifest.yaml [hunk 8]
```diff
@@ -327,19 +362,69 @@ katello_packages:
     rubygem-redhat_access_lib: {}
     rubygem-robotex: {}
     rubygem-runcible: {}
+
 katello_client_packages:
   vars:
     package_base_dir: 'packages/katello/'
     releasers:
     - koji-katello-client
+    repoclosure_config: repoclosure/yum.conf
```
Comment (ekohl) [inline package_manifest.yaml]: Doesn't feel right to group all distros together. Perhaps this is really a limitation of the modeling we chose (and why it's always good to have more than 1 distro - your models are better).

**Comment by komidore64** [comment]
I gave this a smoketest on a couple things. It seemed to run fine.

File: package_manifest.yaml (hunk 8 above)
Comment (ehelms) [reply package_manifest.yaml]: It doesn't feel right but yum/repoclosure is smarter than I even thought. If we don't lump them together, then we have to identify and download each task per distro which is tedious. However, repoclosure if you point it at a directory takes distro into account for us.

**Comment by ehelms** [comment]
@ekohl @komidore64 Updated

File: rel-eng/tito.props [hunk 9]
```diff
@@ -403,26 +403,14 @@ whitelist = ansiblerole-insights-client
...
-
 [katello-nightly-rhel7]
 disttag = .el7
 scl = tfm
 whitelist = katello
   katello-certs-tools
   katello-client-bootstrap
-  katello-host-tools
   katello-installer-base
+  katello-host-tools
```
Comment (ekohl) [inline rel-eng/tito.props]: Why are you moving this? Shouldn't we keep it sorted?

Comment (ehelms) [reply rel-eng/tito.props]: Accidental move -- its back

File: rel-eng/tito.props [hunk 10]
```diff
@@ -403,18 +403,6 @@ whitelist = ansiblerole-insights-client
...
-  katello-repos
-  python-rhsm
-  subscription-manager
-
-[katello-nightly-rhel6]
-disttag = .el6
-whitelist = katello-host-tools
-  katello-repos
-
 [katello-nightly-rhel7]
```
Comment (ekohl) [inline rel-eng/tito.props]: I'm wondering if we should split this in `katello-server-nightly-rhel7` and `katello-client-nightly-rhel7` but that's probably out of scope for this PR.

File: repoclosure/yum.conf [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+
+[el5-subscription-manager]
+name=el5-subscription-manager
+baseurl=https://repos.fedorapeople.org/candlepin/subscription-manager/epel-5/x86_64/
+
+[el5-katello-client-nightly]
+name=Katello nightly EL5
+baseurl=https://fedorapeople.org/groups/katello/releases/yum/nightly/client/el5/x86_64
+
+[f27-base]
```
Comment (ekohl) [inline repoclosure/yum.conf]: Do we need equivalent f28 configs here?

Comment (ehelms) [reply repoclosure/yum.conf]: Not yet -- f28 does not currently work due to a Koji/mock bug.

File: rel-eng/tito.props (hunk 10 above)
Comment (ehelms) [reply rel-eng/tito.props]: I've considered that as well, and out of scope.

**Pull request merged**